### DISCUSSION TITLE
plotly-dash doesn't work

### DISCUSSION THREAD

**Discussion by Vinxenx** [SESMG/SESMG#120] [Q&A] [answered]
Good evening, 
I am in the process of setting up a new laptop. Everything works so far, but I have the problem that the dashboard will not load. Last time the problem could be fixed by installing pip install tool==2.0.3 however I seem to have version 2.2.2. 

her is my recent cmd-process:

C:\Users\vince>cd C:\Users\vince\OneDrive\Dokumente\SESMG-master

C:\Users\vince\OneDrive\Dokumente\SESMG-master>cd scripts

C:\Users\vince\OneDrive\Dokumente\SESMG-master\Scripts>start /b activate.bat

C:\Users\vince\OneDrive\Dokumente\SESMG-master\Scripts>
(SESMG-master) C:\Users\vince\OneDrive\Dokumente\SESMG-master\Scripts>pip freeze
blinker==1.5
Brotli==1.0.9
certifi==2022.6.15
charset-normalizer==2.1.0
click==8.1.3
colorama==0.4.5
cycler==0.11.0
dash==1.7.0
dash-canvas==0.1.0
dash-core-components==1.6.0
dash-html-components==1.0.2
dash-renderer==1.2.2
dash-table==4.5.1
demandlib==0.1.8
dill==0.3.5.1
et-xmlfile==1.1.0
feedinlib==0.0.12
Flask==2.2.2
Flask-Compress==1.12
fonttools==4.34.4
future==0.18.2
graphviz==0.13.2
h5py==3.7.0
idna==3.3
imageio==2.21.1
importlib-metadata==4.12.0
itsdangerous==2.1.2
jdcal==1.4.1
Jinja2==3.1.2
joblib==1.1.0
kiwisolver==1.4.4
MarkupSafe==2.1.1
matplotlib==3.5.3
mock==4.0.3
networkx==2.6.3
nose==1.3.7
numexpr==2.8.3
numpy==1.21.6
oemof.network==0.4.0
oemof.solph==0.4.4
oemof.thermal==0.0.3
oemof.tools==0.4.1
openpyxl==3.0.0
packaging==21.3
pandas==1.3.5
Pillow==9.2.0
plotly==5.9.0
ply==3.11
pvlib==0.7.1
pydot==1.4.1
Pyomo==5.7.1
pyparsing==3.0.9
python-dateutil==2.8.2
pytz==2022.1
PyUtilib==6.0.0
PyWavelets==1.3.0
requests==2.28.1
richardsonpy==0.2.1
scikit-image==0.19.3
scikit-learn==1.0.2
scipy==1.7.3
six==1.16.0
tables==3.5.2
tenacity==8.0.1
threadpoolctl==3.1.0
tifffile==2021.11.2
typing-extensions==4.3.0
urllib3==1.26.11
Werkzeug==2.2.2
windpowerlib==0.0.4
xlrd==1.2.0
XlsxWriter==3.0.3
zipp==3.8.1

C:\Users\vince\OneDrive\Dokumente\SESMG-master\Scripts>pip install werkzeug==2.0.3
Collecting werkzeug==2.0.3
  Downloading https://files.pythonhosted.org/packages/f4/f3/22afbdb20cc4654b10c98043414a14057cd27fdba9d4ae61cea596000ba2/Werkzeug-2.0.3-py3-none-any.whl (289kB)
     |████████████████████████████████| 296kB 3.2MB/s
ERROR: flask 2.2.2 has requirement Werkzeug>=2.2.2, but you'll have werkzeug 2.0.3 which is incompatible.
Installing collected packages: werkzeug
  Found existing installation: Werkzeug 2.2.2
    Uninstalling Werkzeug-2.2.2:
      Successfully uninstalled Werkzeug-2.2.2
Successfully installed werkzeug-2.0.3
WARNING: You are using pip version 19.2.3, however version 22.2.2 is available.
You should consider upgrading via the 'python -m pip install --upgrade pip' command.

(SESMG-master) C:\Users\vince\OneDrive\Dokumente\SESMG-master\Scripts> pip install werkzeug==2.0.3
Requirement already satisfied: werkzeug==2.0.3 in c:\users\vince\onedrive\dokumente\sesmg-master\lib\site-packages (2.0.3)
WARNING: You are using pip version 19.2.3, however version 22.2.2 is available.
You should consider upgrading via the 'python -m pip install --upgrade pip' command.

C:\Users\vince\OneDrive\Dokumente\SESMG-master\Scripts>pip install werkzeug==2.2.3
Collecting werkzeug==2.2.3
  ERROR: Could not find a version that satisfies the requirement werkzeug==2.2.3 (from versions: 0.1, 0.2, 0.3, 0.3.1, 0.4, 0.4.1, 0.5, 0.5.1, 0.6, 0.6.1, 0.6.2, 0.7, 0.7.1, 0.7.2, 0.8, 0.8.1, 0.8.2, 0.8.3, 0.9, 0.9.1, 0.9.2, 0.9.3, 0.9.4, 0.9.5, 0.9.6, 0.10, 0.10.1, 0.10.2, 0.10.4, 0.11, 0.11.1, 0.11.2, 0.11.3, 0.11.4, 0.11.5, 0.11.6, 0.11.7, 0.11.8, 0.11.9, 0.11.10, 0.11.11, 0.11.12, 0.11.13, 0.11.14, 0.11.15, 0.12, 0.12.1, 0.12.2, 0.13, 0.14, 0.14.1, 0.15.0, 0.15.1, 0.15.2, 0.15.3, 0.15.4, 0.15.5, 0.15.6, 0.16.0, 0.16.1, 1.0.0rc1, 1.0.0, 1.0.1, 2.0.0rc1, 2.0.0rc2, 2.0.0rc3, 2.0.0rc4, 2.0.0rc5, 2.0.0, 2.0.1, 2.0.2, 2.0.3, 2.1.0, 2.1.1, 2.1.2, 2.2.0a1, 2.2.0, 2.2.1, 2.2.2)
ERROR: No matching distribution found for werkzeug==2.2.3
WARNING: You are using pip version 19.2.3, however version 22.2.2 is available.
You should consider upgrading via the 'python -m pip install --upgrade pip' command.

(SESMG-master) C:\Users\vince\OneDrive\Dokumente\SESMG-master\Scripts>pip install werkzeug==22.2.2
Collecting werkzeug==22.2.2
  ERROR: Could not find a version that satisfies the requirement werkzeug==22.2.2 (from versions: 0.1, 0.2, 0.3, 0.3.1, 0.4, 0.4.1, 0.5, 0.5.1, 0.6, 0.6.1, 0.6.2, 0.7, 0.7.1, 0.7.2, 0.8, 0.8.1, 0.8.2, 0.8.3, 0.9, 0.9.1, 0.9.2, 0.9.3, 0.9.4, 0.9.5, 0.9.6, 0.10, 0.10.1, 0.10.2, 0.10.4, 0.11, 0.11.1, 0.11.2, 0.11.3, 0.11.4, 0.11.5, 0.11.6, 0.11.7, 0.11.8, 0.11.9, 0.11.10, 0.11.11, 0.11.12, 0.11.13, 0.11.14, 0.11.15, 0.12, 0.12.1, 0.12.2, 0.13, 0.14, 0.14.1, 0.15.0, 0.15.1, 0.15.2, 0.15.3, 0.15.4, 0.15.5, 0.15.6, 0.16.0, 0.16.1, 1.0.0rc1, 1.0.0, 1.0.1, 2.0.0rc1, 2.0.0rc2, 2.0.0rc3, 2.0.0rc4, 2.0.0rc5, 2.0.0, 2.0.1, 2.0.2, 2.0.3, 2.1.0, 2.1.1, 2.1.2, 2.2.0a1, 2.2.0, 2.2.1, 2.2.2)
ERROR: No matching distribution found for werkzeug==22.2.2
WARNING: You are using pip version 19.2.3, however version 22.2.2 is available.
You should consider upgrading via the 'python -m pip install --upgrade pip' command.

C:\Users\vince\OneDrive\Dokumente\SESMG-master\Scripts>

Thank u already!

Kind regards vincent

**Comment by GregorBecker** [comment]
Good evening @Vinxenx, 

the guess resetting "Werkzeug" to 2.0.3 is actually the correct one. However, you have tried numerous other versions. Is there a reason for this? 
Can you set the version again to 2.0.3 and send us the error message when creating the plotly dash?

Best regards 

Gregor

**Reply by Vinxenx** [reply]
Good evening gregor,

i tried to fix it on my own. This is why I tried other versions. It is telling me that the 2.0.3 is still installed. 

C:\Users\vince\OneDrive\Dokumente\SESMG-master\Scripts>pip install werkzeug==2.0.3
Requirement already satisfied: werkzeug==2.0.3 in c:\users\vince\onedrive\dokumente\sesmg-master\lib\site-packages (2.0.3)
WARNING: You are using pip version 19.2.3, however version 22.2.2 is available.
You should consider upgrading via the 'python -m pip install --upgrade pip' command.

(SESMG-master) C:\Users\vince\OneDrive\Dokumente\SESMG-master\Scripts>

In the png u can see the error.

However here is also what the cmd from sesmg is showing me:

18:08:01-INFO-   Starting Optimization with cbc-Solver
18:09:57-INFO-Optimization successful...
18:10:15-INFO-   Successfully prepared results...
18:10:15-INFO-   --------------------------------------------------------
18:10:15-INFO-   Modelling and optimization successfully completed!
Traceback (most recent call last):
  File "program_files/Interactive_Results.py", line 18, in <module>
    import dash
  File "C:\Users\vince\OneDrive\Dokumente\SESMG-master\lib\site-packages\dash\__init__.py", line 1, in <module>
    from .dash import Dash, no_update  # noqa: F401
  File "C:\Users\vince\OneDrive\Dokumente\SESMG-master\lib\site-packages\dash\dash.py", line 19, in <module>
    import flask
  File "C:\Users\vince\OneDrive\Dokumente\SESMG-master\lib\site-packages\flask\__init__.py", line 4, in <module>
    from . import json as json
  File "C:\Users\vince\OneDrive\Dokumente\SESMG-master\lib\site-packages\flask\json\__init__.py", line 8, in <module>
    from ..globals import current_app
  File "C:\Users\vince\OneDrive\Dokumente\SESMG-master\lib\site-packages\flask\globals.py", line 57, in <module>
    _cv_app, unbound_message=_no_app_msg
TypeError: __init__() got an unexpected keyword argument 'unbound_message'

![image](https://user-images.githubusercontent.com/96150761/184227186-4a048d7c-9bfb-4a83-80e7-739ded508bfb.png)

Best regards vincent

**Comment by GregorBecker** [comment, marked as answer]
Thanks @Vinxenx, I'll take a look at it tomorrow, try to reproduce it and get back to you. I hope and think, that we can find a quick way to solve your problem.

**Reply by Vinxenx** [reply]
Have a good evening,  thank u for the great support Gregor. :)

**Reply by GregorBecker** [reply]
Good morning @Vinxenx, 

I was able to reproduce and fix your problem. 

You have to downgrade the "Werkzeug" version to 2.0.3: 

`pip install Werkzeug==2.0.3`

and unfortunately also downgrade the "Flask" version to 2.1.3:

`pip install flask==2.1.3`

I hope it works, if not please contact me. 

Kind regards 

Gregor

**Reply by Vinxenx** [reply]
It worked! Thank you!
